### ISSUE TITLE
NoSuchMethodError

### ISSUE THREAD

**Issue by Allsimon** [simasch/vaadin-jooq#3]
Hello,

I just found your project and tried to use it but I'm very puzzled by a `NoSuchMethodError`:

```
Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Servlet execution threw an exception] with root cause

java.lang.NoSuchMethodError: 'org.jooq.SelectWhereStep org.jooq.DSLContext.selectFrom(org.jooq.Table)'
	at io.seventytwo.vaadinjooq.repository.JooqRepository.findAll(JooqRepository.java:47) ~[vaadin-jooq-1.2.0.jar:na]
```

I'm using jOOQ 3.17.5, the version is the same everywhere when checking in `mvn dependency:tree`, I re-generated the code a few times to be sure

If I do an "Evaluate expression" of your code in IntelliJ, it runs fine... but stepping over the line throws this exception.

I just created this bean in the same application and it works fine:
```java
@Service
public record Test(DSLContext jooq) {

    @PostConstruct
    public void test() {
        System.out.println(jooq.selectFrom(MY_TABLE).fetch());
    }
}
```

I have absolutely no idea wtf is going on, did you encounter this issue too ?

**Comment by Allsimon** [comment]
I git cloned this project, bumped jOOQ to 3.17.5, `mvn clean install` then used the snapshot in my project and it fixed the issue.

Apparently the versions need to be aligned... even though your pom.xml declares it correctly as "provided" and manually forcing with exclusion does not work 🤔 

I'm closing this issue since I found a workaround, but if you have an idea why it's behaving this way, I'm very interested

**Comment by simasch** [comment]
Hi @Allsimon 
Thanks for reporting the issue.

The method signature has changed. I will create a new version that is compatible with jOOQ 3.17

**Comment by simasch** [comment]
jOOQ 3.17 is not binary compatible with jOOQ 3.16 and it also requires Java 17.

I've released a new version 2.0.1. Please use this when using jOOQ 3.17.
